lib/github.ml

```diff
@@ -235,6 +250,61 @@ let assert_tag_exists ~dry_run tag =
   if Vcs.tag_exists ~dry_run repo tag then Ok ()
   else R.error_msgf "%s is not a valid tag" tag
 
+(* Ask the user then push the tag. Guess the ssh URI from the dev-repo.
+   This function can abort:
+   - The user answered "N" to pushing the tag
+   - The push failed
+
+   This function does nothing if the tag is already present on the remote and
+   point to the same ref. *)
+let push_tag ~dry_run ~yes ~dev_repo vcs tag =
+  let remote_has_tag_uptodate () =
+    if dry_run then Ok true (* Assume yes when [dry_run] is enabled *)
+    else
+      Vcs.commit_id ~dirty:false ~commit_ish:tag vcs >>= fun local_rev ->
+      Vcs.ls_remote ~dry_run vcs ~kind:`Tag ~filter:tag dev_repo >>= function
+      | [] -> Ok false
+      | (rev, _) :: _ when local_rev = rev -> Ok true
+      | (rev, _) :: _ ->
+          App_log.unhappy (fun l ->
+              l
+                "The tag %a is present on the remote but point to a different \
+                 commit (%a)."
+                Text.Pp.version tag Text.Pp.commit rev);
+          Ok false
+  in
+  remote_has_tag_uptodate () >>= function
+  | true -> Ok () (* No need to push, avoiding the need to guess the uri. *)
+  | false -> (
+      let uri =
+        match Parse.ssh_uri_from_http dev_repo with
+        | Some uri -> uri
+        | None ->
+            App_log.unhappy (fun l ->
+                l
+                  "The uri %a is not recognized as a gihub uri, we are going \
+                   to assume it is already a ssh uri."
+                  Text.Pp.url dev_repo);
+            dev_repo
```

Comment:
This is the only place we interpret the `dev_repo` field, the rest of the function uses the `distrib_uri`.
This is the reason for this PR: The guess is wrong sometimes. This PR adds a way to skip this step by pushing the tag manually.

Reply:
I was asking due to the following reason: there are two possibilities:
1. The only way to get into this part of the code is that the guess is wrong. I think in that case it makes sense to report that with `App_log.unhappy` (although in that case we might as well abort directly instead of keeping on going). 
2. It is also possible to get into this part of the code and afterwards the command will still succeed. I was wondering if in that case it might be better to report that with `App_log.status`. What do you think?

I think if dev-repo is of the form ` ssh://git@github.com/...` we might be in possibility 2 (but I'm not sure if people ever use that format, since it's a bit strange). 

Reply:
I think it makes sense to keep going. This URI is already meant to be interpreted by Git and that check is only here to work around a common problem: Dev-repo is a Github's HTTP uri but we want to push using Github's SSH uris.
Note that this was already the case and this PR only adds messages and moves code around.

Reply:
Ok. I was more talking about the kind of reporting (`unhappy` vs `status`), but I'm not sure myself, so leaving it is good for me.